js/my-github.js

```diff
@@ -1,2 +1,83 @@
 // Write code here to communicate with Github
 
+//Part-1
+// var url = "https://api.github.com/users/nabraha/repos";
+// fetch(url)
+//   .then(resp => resp.json())
+//   .then(function(result) {
+//     numberOfRepos(result);
+//     result.forEach(function(name) {
+//       reposListAndLink(name);
+//     });
+//   });
+
+// function reposListAndLink(name) {
+//   const getRepoId = document.getElementById("repos-list");
+//   const repoItemElement = document.createElement("li");
+//   repoItemElement.appendChild(addAnchor(name));
+//   getRepoId.appendChild(repoItemElement);
+// }
+
+// function addAnchor(name) {
+//   var addLink = document.createElement("a");
+//   addLink.href = name.html_url;
+//   addLink.text = name.name;
+//   return addLink;
+// }
+
+// function numberOfRepos(result) {
+//   var repoNumber = document.getElementById("repos-count");
+//   repoNumber.innerText = result.length;
+//   //  repoNumber.innerText = Object.keys(result).length;
+// }
+
+//part-3
+
+var repoNumber = document.getElementById("repos-count");
+const getRepoId = document.getElementById("repos-list");
+const getForm = document.querySelector(".my-auto form:nth-child(4)");
+const defaultUser = "Nabraha";
+getForm.addEventListener("submit", function(e) {
+  e.preventDefault();
+
+  var value = getForm.querySelector('input[type="text"]').value.toLowerCase();
+  if (value === "") {
+    value = defaultUser;
+  }
+
+  var url = "https://api.github.com/users/" + value + "/repos";
+  fetch(url)
+    .then(resp => resp.json())
+    .then(function(result) {
+      result.forEach(function(name) {
```

Comment:
What is `name` here? I would avoid calling something name unless it actually represents a name. In Javascript we don't have strict types to guide us, so we rely on naming to know what something is. So result could be `repos` and `name` could be `repo`.